README.md

```diff
@@ -12,66 +12,88 @@ And I guess I'm just yet another one looking at the xkcd:File Transfer thing:
 
 Other (nicer) alternatives exist, web based (WebRTC in fact) (even serverless [like this one] (http://blog.printf.net/articles/2013/05/17/webrtc-without-a-signaling-server)), but I wanted to go for a command line solution. Also, most p2p options need a central server to do the exchange of the public IPs before starting the "hole punching". I also wanted to avoid this, so the exchange is done manually, sharing the public IPs using your favourite messaging platform (like Slack).
 
-It is built on top of UDT, the famous library to speed up data transfer on high bandwidth, high latency networks. It includes a "rendezvous" mode to perform UDP hole punching, and that's what I use.
+To perform UDP hole punching, custom punch packets are exchanged before the TCP handshake.
 
 # How to use it
-The two peers will need a copy of p2p.exe, then one will act as "sender" and the other as "receiver" (in fact, using the commands with these names).
+The two peers will need a copy of p2pcopy.exe, then one will act as "sender" and the other as "receiver" (in fact, using the commands with these names).
 
 ## Sender
-I'm specifiying a local port, which is not mandatory, you can skip the --localport.
-
+I'm specifying a local port, which is not mandatory, you can skip the --localport. In this example, the sender is on AWS and non-NATted, the receiver is on a home router and NATted.
```

Comment:
wouldn't it be better to put an AWSless example?

Reply:
It would be! But am not currently set up to test NATted to NATted. Should work though.

Reply:
@psantosl please test NATted to NATted before I release bounty (and if it works, let's modify this to be AWSless).

BTW why are you not currently set up? wouldn't it be as simple as having 2 computers, one connected via wifi, and other one connected via the WiFi-hotspot of a mobile device?

Reply:
@knocte The standard UDP hole punching technique doesn't work on mobile data, because mobile networks use symmetric NAT, which means the port number changes with the destination even if you reuse the socket. [This paper goes into the details](https://drive.google.com/file/d/0B1IimJ20gG0SY2NvaE4wRVVMbG8/view). The TUM NatAnalyzer is unable to achieve NAT traversal when I use mobile data. Would be extremely surprised if the original version of p2pcopy worked either -- will test though.